src/app/newSettings/components/BillingPaymentMethodCard.tsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useState } from 'react';
+import { t } from 'i18next';
+import { PaymentMethod } from '@internxt/sdk/dist/drive/payments/types';
+import { Source } from '@stripe/stripe-js';
+
+import paymentService from 'app/payment/services/payment.service';
+
+import Spinner from 'app/shared/components/Spinner/Spinner';
+import Button from 'app/shared/components/Button/Button';
+import Card from 'app/shared/components/Card';
+
+import visaIcon from '../../../assets/icons/card-brands/visa.png';
+import amexIcon from '../../../assets/icons/card-brands/amex.png';
+import dinersIcon from '../../../assets/icons/card-brands/diners_club.png';
+import discoverIcon from '../../../assets/icons/card-brands/discover.png';
+import jcbIcon from '../../../assets/icons/card-brands/jcb.png';
+import mastercardIcon from '../../../assets/icons/card-brands/mastercard.png';
+import unionpayIcon from '../../../assets/icons/card-brands/unionpay.png';
+import unknownIcon from '../../../assets/icons/card-brands/unknown.png';
+import EditPaymentMethodModal from './EditPaymentMethodModal';
+
+interface StateProps {
+  tag: 'ready' | 'loading' | 'empty';
+  card?: PaymentMethod['card'];
+  type?: Source.Type;
+}
+
+const BillingPaymentMethodCard = () => {
+  const [isEditPaymentMethodModalOpen, setIsEditPaymentMethodModalOpen] = useState<boolean>(false);
+  const [state, setState] = useState<StateProps>({
+    tag: 'loading',
+  });
+
+  const cardBrands: Record<PaymentMethod['card']['brand'], string> = {
+    visa: visaIcon,
+    amex: amexIcon,
+    diners: dinersIcon,
+    discover: discoverIcon,
+    jcb: jcbIcon,
+    mastercard: mastercardIcon,
+    unionpay: unionpayIcon,
+    unknown: unknownIcon,
+  };
+
+  const paymentsTypes = {
+    paypal: 'PayPal',
+    sepa_debit: 'SEPA Direct Debit',
+  };
+
+  useEffect(() => {
+    paymentService
+      .getDefaultPaymentMethod()
+      .then((data: PaymentMethod | Source) => {
+        if (data.card) {
+          setState({ tag: 'ready', card: data.card as PaymentMethod['card'] });
+        } else if ('type' in data) {
+          setState({ tag: 'ready', type: data.type });
+        } else {
+          setState({ tag: 'empty' });
+        }
+      })
+      .catch(() => setState({ tag: 'empty' }));
+  }, []);
```

Comment:
Same as before, I guess with this component is the one that was in the old Settings. You can create a container that implements this call and so we take it out of the component and it is ready to move it to the library.

Reply:
Done